### PR TITLE
Generate default impl when converting #[derive(Default)] to manual impl

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Similar to https://github.com/rust-analyzer/rust-analyzer/pull/9814, but for `#[derive(Default)]`. Thanks!

## Follow-up steps
I've added the tests inside `handlers/replace_derive_with_manual_impl.rs` again, but I'm planning a follow-up PR to extract these to `utils/` so we can share them between assists - and maybe even add another assist just for the purpose of testing these impls (e.g. `generate_default_trait_body`).

The step after _that_ is likely to fill out the remaining traits, so we can make it so whenever RA auto-completes a trait which also can be derived, we provide a default function body.

### CODE REVIEW

File: crates/test_utils/src/minicore.rs [hunk 1]
```diff
@@ -346,6 +346,14 @@ pub mod fmt {
 }
 // endregion:fmt
 
+// region:default
```
Comment (lnicola) [inline crates/test_utils/src/minicore.rs:349]: Isn't this already there at line 82?

Comment (yoshuawuyts) [reply crates/test_utils/src/minicore.rs:349]: 😬 

fixing now!

**Review by lnicola — APPROVED** [review]
Looks fine, I guess, but the code structure (`return match`, and `Some(())` and `return Some();` instead of `None` which would have made more sense) is a bit strange.

**Comment by yoshuawuyts** [comment]
Oh I can change it to `None`, and move the function definition to the top of the function body. Let me do both of those real quick!

**Review by lnicola — APPROVED** [review]

**Comment by lnicola** [comment]
If you want, you can record a GIF to showcase this off (and maybe hit Save faster), but it's fine too if you don't.

**Comment by yoshuawuyts** [comment]
@lnicola I might do a few more PRs throughout the week. Perhaps it would make sense to generate the gifs closee to the end of the week, so we can think about the larger picture then (:

**Comment by lnicola** [comment]
generate-impls

**Pull request merged**